src/responder/common/responder_common.c

```diff
@@ -549,6 +555,9 @@ static void accept_fd_handler(struct tevent_context *ev,
         return;
     }
 
+    /* Set the chain id */
+    cctx->old_chain_id = sss_chain_id_set(rctx->client_id_num);
```

Comment:
I'm not sure I understand how it works.

If you set chain-id in `accept_fd_handler()` then new requests from all currently connected clients will have the same CID (of last connected client), no?

Wouldn't it be better to set chain-id somewhere "around" `client_fd_handler()`?

Reply:
When setting the chain ID (and incrementing the id number) inside client_fd_handler() functions it causes every new cache request to increment the ID, rendering the client tracking ineffective. That is why incrementing the id number is done in accept_fd_handler() 

Yes there will be a downside of duplciate IDs for any client or daemon which runs multiple commands over a single socket connection, but IIUC we agreed to accept this deficiency for now - knowing that ID tracking is often useful to assist in troubleshooting user run command tests.

Reply:
I don't question incrementing in `accept_fd_handler()`.
I question `sss_chain_id_set()` there.

Consider a case of multiple client connected (and making requests) simultaneously - those needs to be distinguished.

Reply:
Okay, I moved `sss_chain_id_set` into `sss_client_fd_handler()`. Thank you